modules/linear_boltzmann_solvers/response_evaluator/response_evaluator.cc

```diff
@@ -0,0 +1,468 @@
+              const auto& node = grid.vertices[cell.vertex_ids_[i]];
+              const auto& intF_shapeI = fe_values.intS_shapeI[f][i];
+
+              const auto psi_bndry = EvaluateBoundaryCondition(bndry_id, node, groupset);
+
+              for (size_t n = 0; n < num_gs_angles; ++n)
+              {
+                const auto& omega = quadrature->omegas_[n];
+                const auto mu = omega.Dot(face.normal_);
+                if (mu < 0.0)
+                {
+                  const auto& wt = quadrature->weights_[n];
+                  const auto weight = -mu * wt * intF_shapeI;
+                  const auto dof_map = discretization.MapDOFLocal(cell, i, uk_man, n, 0);
+
+                  for (size_t gsg = 0; gsg < num_gs_groups; ++gsg)
+                    local_response +=
+                      weight * psi_dagger[gs][dof_map + gsg] * psi_bndry[num_gs_groups * n + gsg];
+                } // if outgoing
+              }
+            } // for face node fi
+          }
+          ++f;
+        } // for face
+      }   // for cell
+      ++gs;
+    } // for groupset
+  }   // if boundary sources
+
+  // Point sources
+  for (const auto& point_source : point_sources_)
+    for (const auto& subscriber : point_source.Subscribers())
+    {
+      const auto& cell = grid.local_cells[subscriber.cell_local_id];
+      const auto& transport_view = transport_views[cell.local_id_];
+
+      const auto& src = point_source.Strength();
+      const auto& vol_wt = subscriber.volume_weight;
+
+      const auto num_cell_nodes = transport_view.NumNodes();
+      for (size_t i = 0; i < num_cell_nodes; ++i)
+      {
+        const auto dof_map = transport_view.MapDOF(i, 0, 0);
+        const auto& shape_val = subscriber.shape_values[i];
+        for (size_t g = 0; g < num_groups; ++g)
+          local_response += vol_wt * shape_val * src[g] * phi_dagger[dof_map + g];
+      } // for node i
+    }   // for subscriber
+
+  // Distributed sources
+  for (const auto& distributed_source : distributed_sources_)
+    for (const uint64_t local_id : distributed_source.Subscribers())
+    {
+      const auto& cell = grid.local_cells[local_id];
+      const auto& transport_view = transport_views[cell.local_id_];
+      const auto& fe_values = unit_cell_matrices[cell.local_id_];
+      const auto& nodes = discretization.GetCellNodeLocations(cell);
+
+      const auto num_cell_nodes = transport_view.NumNodes();
+      for (size_t i = 0; i < num_cell_nodes; ++i)
+      {
+        const auto& V_i = fe_values.intV_shapeI[i];
+        const auto dof_map = transport_view.MapDOF(i, 0, 0);
+        const auto& vals = distributed_source(cell, nodes[i], num_groups);
+        for (size_t g = 0; g < num_groups; ++g)
+          local_response += vals[g] * phi_dagger[dof_map + g] * V_i;
+      }
+    }
+
+  double global_response = 0.0;
+  MPI_Allreduce(&local_response, &global_response, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
```

Comment:
This should be:
```
mpi_comm.all_reduce(local_response, global_response, mpi::op::sum<double>());
```
And somewhere at the top of the file, you will need to add:
```
#include "mpicpp-lite/mpicpp-lite.h"

namespace mpi = mpicpp_lite;
```

I know I commented on this, but I cannot find the comment, so I must have moved pages in the browser and the comment got lost in the first review. Sorry about that.